sim/pokemon.ts

```diff
@@ -193,8 +193,13 @@ export class Pokemon {
 	 * always be overwritten by a move action before the end of that turn.
 	 */
 	moveThisTurnResult: boolean | null | undefined;
-	/** used for Assurance check */
-	hurtThisTurn: boolean;
+	/**
+	 * The undynamaxed HP value this Pokemon was reduced to by damage this turn,
+	 * or false if it hasn't taken damage yet this turn
+	 *
+	 * Used for Assurance, Emergency Exit, and Wimp Out
+	 */
+	hurtThisTurn: number | false;
```

Comment:
Is there any mileage in having this be zero instead of false?

Reply:
Functionally it would be pretty much the same either way, but `false` is easier to understand as “was not hurt this turn” than `0` is, in my opinion. This is a sort of hybrid property used for two admittedly somewhat unrelated checks, so I think it’s fitting for its type to be hybrid as well.

Reply:
I just thought the extra cast looked odd that's all (I don't know if it's a common thing to do in PS! code).

Reply:
CONTRIBUTING specifies that this should be `number | null`, not `number | false`. Can you make that change?